### PR TITLE
Endgame

### LINKED ISSUE #182: Need to end the game in some way
**What Do You Want?**
We need logic to end a game.

- Do we go for a fixed number of rounds/ till 1 player is left?
- What happens if no one gets a space in a round? Does the game end with no winners or do we have joint winners?

### PR DESCRIPTION
**Pull request by ChrisGora**
The `GameEnd` event now includes a list of winners. The GUI will inform you if you won based on this list. The `Eliminate` event already included a list of eliminated players. The GUI will now inform you if you have been eliminated.

Resolves #182

### CODE REVIEW

**Review by Keywarn — DISMISSED** [review]
<3

File: ProjectFiles/Assets/Scripts/Game/RoundManager.cs [hunk 1]
```diff
@@ -117,15 +121,16 @@ private void EndRoundEvent()
         {
             NotifyRoundEnd();
             roundNumber++;
-            if (roundNumber < maxRounds)
+            if (roundNumber < RoundProperties.MaxRounds && world.GetNumPlayers() > 1)
```
Comment (RileyEv) [inline ProjectFiles/Assets/Scripts/Game/RoundManager.cs]: This logic isn't going to work. The world.GetNumPlayers won't go below 1 until the players have been eliminated, so whatever happens there will have to be one more round with just 1 player before the game ends.

File: ProjectFiles/Assets/Scripts/Game/RoundManager.cs [hunk 2]
```diff
@@ -101,14 +98,21 @@ public void StartPreRound()
 
         private void StartRoundEvent()
         {
-            // TODO: give an actual dynamic value to num space
+            int numberOfSpaces = 1;
+            // If not the last round
+            if (roundNumber < RoundProperties.MaxRounds - 1)
+            {
+                numberOfSpaces = (int) Math.Ceiling(world.GetNumPlayers() * RoundProperties.SpacesToPlayersRatio);
```
Comment (RileyEv) [inline ProjectFiles/Assets/Scripts/Game/RoundManager.cs]: Doesn't work with 2 players. 

ceil(2 * 0.6) = 2 so there's always two spaces for two player
```suggestion
                numberOfSpaces = (int) Math.Floor(world.GetNumPlayers() * RoundProperties.SpacesToPlayersRatio);
```

**Review by RileyEv — CHANGES_REQUESTED** [review]
Some of the logic doesn't work. We also never finished the work on `disconnect-hotfix` hence why it wasn't merged into master, so disconnecting of players also doesn't work.

**Comment by ChrisGora** [comment]
Either change it rn or I can try to change it tomorrow at some point before 2pm

**Comment by ChrisGora** [comment]
Gonna fix shit now

**Comment by RileyEv** [comment]
I’ll be in uni for just after 11 so when your done come in and we can test it

Many Thanks

Riley Evans

**Comment by ChrisGora** [comment]
maybe

Warmest Regards,
Krzysztof

**Comment by RileyEv** [comment]
It was my automatic signature in my email 🙈

Many Thanks

Riley Evans

**Comment by ghost** [comment]
good luck at the demo today

Season's greetings,
Jack Bond-Preston

**Comment by RileyEv** [comment]
Good luck in your interview today

Many thanks,

Riley Evans
 www.rileyevans.co.uk

**Review by RileyEv — APPROVED** [review]

**Pull request merged**